skyportal/enum_types.py

```diff
@@ -1,16 +1,38 @@
-import sqlalchemy as sa
-from enum import Enum
+import astropy.units as u
 import inspect
+import numpy as np
+import sncosmo
+from sncosmo.bandpasses import _BANDPASSES
+from sncosmo.magsystems import _MAGSYSTEMS
+import sqlalchemy as sa
 
+from enum import Enum
 from baselayer.app.env import load_env
-
+from baselayer.log import make_log
 from . import facility_apis
 
-from sncosmo.bandpasses import _BANDPASSES
-from sncosmo.magsystems import _MAGSYSTEMS
+log = make_log('enum_types')
 
 _, cfg = load_env()
 
+# load additional bandpasses into the SN comso registry
+additional_bandpasses_names = []
+for additional_bandpasses in cfg.get('additional_bandpasses', []):
+    name = additional_bandpasses.get("name")
+    if not name:
+        continue
+    try:
+        wavelength = np.array(additional_bandpasses.get("wavelength"))
+        transmission = np.array(additional_bandpasses.get("transmission"))
+    except Exception as e:
+        log(f"Could not make bandpass for {name}")
+        log(f"(Original exception: {e})")
+
+    band = sncosmo.Bandpass(wavelength, transmission, name=name, wave_unit=u.AA)
```

Comment:
Shouldn't the sncosmo.Bandpass call also be wrapped in the try except? Perhaps there should be some kind of check specifically for the names not conflicting with existing built-ins. Do you mean for there to be a continue in the except? And maybe if not name should also print some kind of warning?

Reply:
Good points (and good catch on the continue). All fixed.